scenarios/lambda_injection_privesc/terraform/lambda_source_code/policy_applier_lambda_src/main.py

```diff
@@ -0,0 +1,51 @@
+import boto3
+from sqlite_utils import Database
+
+db = Database("my_database.db")
+iam_client = boto3.client('iam')
+
+# db["policies"].insert_all([
+#     {"policy_name": "AmazonSNSReadOnlyAccess", "public": 'True'}, 
+#     {"policy_name": "AmazonRDSReadOnlyAccess", "public": 'True'},
+#     {"policy_name": "AWSLambda_ReadOnlyAccess", "public": 'True'},
+#     {"policy_name": "AmazonS3ReadOnlyAccess", "public": 'True'},
+#     {"policy_name": "AmazonGlacierReadOnlyAccess", "public": 'True'},
+#     {"policy_name": "AmazonRoute53DomainsReadOnlyAccess", "public": 'True'},
+#     {"policy_name": "AdministratorAccess", "public": 'False'}
+# ])
+
+
+def handler(event, context): 
+    target_policys = event['policy_names']
+    user_name = event['user_name']
+    print(f"target policys are : {target_policys}")
+
+    for policy in target_policys:
+        statement_returns_valid_policy = False
+        statement = f"select policy_name from policies where policy_name='{policy}' and public='True'"
+        for row in db.query(statement):
+            statement_returns_valid_policy = True
+            print(f"applying {row['policy_name']} to {user_name}")
+            response = iam_client.attach_user_policy(
+                UserName= user_name,
+                PolicyArn=f"arn:aws:iam::aws:policy/{row['policy_name']}"
+                )
```

Comment:
Unsure how realistic this part is, it works fine though and get's the point across so I'm not this is terribly important.

Basically there's not much of a reason to use SQL here since this could just be a python list of approved policies. I think a case that is similar and maybe more likely is if these where custom policies and the DB had two columns, one for the name and one for the custom policy. 

The query would then just be returning the policy that should be applied to the user and the SQL injection should return an arbitrary policy. So if db.query allows multiple statements I think this would just mean terminating the first query so there is no output then using `SELECT '<policy to apply>'`.

Up to you though if you want to do this or not, think either way will work.

Reply:
@RyanJarv I think this makes a lot of sense, and I like the concept. What do you think about adding a second lambda at some point in the future so there are two paths? If there is a downside to having an unrealistic injection scenario I'm open to replacing the current lambda too.

Reply:
This idea is valuable, but I was not able to get a more realistic injection scenario to work. That work still exists in the form of commented out terraform resources. I might look into building a second lambda that has a code injection vuln in the future.

Resolving this conversation for the moment.